specs/networking/p2p-interface.md

```diff
@@ -0,0 +1,743 @@
+
+**Protocol ID:** `/meshsub/1.0.0`
+
+**Gossipsub Parameters**
+
+*Note: Parameters listed here are subject to a large-scale network feasibility study.*
+
+The following gossipsub parameters will be used:
+
+- `D` (topic stable mesh target count): 6
+- `D_low` (topic stable mesh low watermark): 4
+- `D_high` (topic stable mesh high watermark): 12
+- `D_lazy` (gossip target): 6
+- `fanout_ttl` (ttl for fanout maps for topics we are not subscribed to but have published to, seconds): 60
+- `gossip_advertise` (number of windows to gossip about): 3
+- `gossip_history` (number of heartbeat intervals to retain message IDs): 5
+- `heartbeat_interval` (frequency of heartbeat, seconds): 1
+
+### Topics
+
+Topics are plain UTF-8 strings, and are encoded on the wire as determined by protobuf (gossipsub messages are enveloped in protobuf messages).
+
+Topic strings have form: `/eth2/TopicName/TopicEncoding`. This defines both the type of data being sent on the topic and how the data field of the message is encoded. (Further details can be found in [Messages](#Messages)).
+
+There are two main topics used to propagate attestations and beacon blocks to all nodes on the network. Their `TopicName`'s are:
+
+- `beacon_block` - This topic is used solely for propagating new beacon blocks to all nodes on the networks. Blocks are sent in their entirety. Clients who receive a block on this topic MUST validate the block proposer signature before forwarding it across the network.
+- `beacon_attestation` - This topic is used to propagate aggregated attestations (in their entirety) to subscribing nodes (typically block proposers) to be included in future blocks. Similarly to beacon blocks, clients will be expected to perform some sort of validation before forwarding, but the precise mechanism is still TBD.
+
+Additional topics are used to propagate lower frequency validator messages. Their `TopicName`’s are:
+
+- `voluntary_exit` - This topic is used solely for propagating voluntary validator exits to proposers on the network. Voluntary exits are sent in their entirety. Clients who receive a voluntary exit on this topic MUST validate the conditions within `process_voluntary_exit` before forwarding it across the network.
+- `proposer_slashing` - This topic is used solely for propagating proposer slashings to proposers on the network. Proposer slashings are sent in their entirety. Clients who receive a proposer slashing on this topic MUST validate the conditions within `process_proposer_slashing` before forwarding it across the network.
+- `attester_slashing` - This topic is used solely for propagating attester slashings to proposers on the network. Attester slashings are sent in their entirety. Clients who receive an attester slashing on this topic MUST validate the conditions within `process_attester_slashing` before forwarding it across the network.
+
+#### Interop
+
+Unaggregated attestations from all shards are sent to the `beacon_attestation` topic.
```

Comment:
What about aggregated attestations? They also go to `beacon_attestation` topic in interop config?

Reply:
For lighthouse, during initial interop, we were planning on not doing aggregation (just to simplify initial compatibility). As interop progresses I think yes, we should be sending aggregated attestations to `beacon_attestation`.

Reply:
Indeed, the aggregation strategy has yet to be decided. An open question is whether to allow aggregated attestations here explicitly - it's fine for nimbus, but would require that all testnet participants can at least deal with reading aggregated attestations. Shout if this is a problem, or we'll allow aggregated also.

Reply:
>  it's fine for nimbus, but would require that all testnet participants can at least deal with reading aggregated attestations

these are the exact some datastructure so I think readability should be fine. I lean toward being able to broadcast aggregates on this same channel for interop

> the aggregation strategy has yet to be decided.

I need to get the "naive" strategy in the validator (or network) spec asap. (subset of committee aggregates and broadcasts to `beacon_`  channel)

Reply:
> I need to get the "naive" strategy in the validator (or network) spec asap.

It's not a blocker for interop, but indeed a high-priority concern so we can start investigating performance. 

https://github.com/ethereum/eth2.0-specs/issues/1331